terraform/performance/main.tf

```diff
@@ -74,12 +74,13 @@ resource "null_resource" "upload-validator" {
 #####################################################################
 resource "aws_instance" "cwagent" {
 
-  ami                         = data.aws_ami.latest.id
-  instance_type               = var.ec2_instance_type
-  key_name                    = local.ssh_key_name
-  iam_instance_profile        = module.basic_components.instance_profile
-  vpc_security_group_ids      = [module.basic_components.security_group]
-  associate_public_ip_address = true
+  ami                                  = data.aws_ami.latest.id
+  instance_type                        = var.ec2_instance_type
+  key_name                             = local.ssh_key_name
+  iam_instance_profile                 = module.basic_components.instance_profile
+  vpc_security_group_ids               = [module.basic_components.security_group]
+  associate_public_ip_address          = true
+  instance_initiated_shutdown_behavior = "terminate"
```

Comment:
No reason not to terminate EC2 instances on shutdown, right?

Reply:
>[instance_initiated_shutdown_behavior](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#instance_initiated_shutdown_behavior) - (Optional) Shutdown behavior for the instance. Amazon defaults this to stop for EBS-backed instances and terminate for instance-store instances. Cannot be set on instance-store instances. See [Shutdown Behavior](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/terminating-instances.html#Using_ChangingInstanceInitiatedShutdownBehavior) for more information.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance